libp2p/Host.cpp

```diff
@@ -167,94 +172,50 @@ void Host::registerPeer(std::shared_ptr<Session> _s, CapDescs const& _caps)
 		}
 }
 
-void Host::seal(bytes& _b)
-{
-	_b[0] = 0x22;
-	_b[1] = 0x40;
-	_b[2] = 0x08;
-	_b[3] = 0x91;
-	uint32_t len = (uint32_t)_b.size() - 8;
-	_b[4] = (len >> 24) & 0xff;
-	_b[5] = (len >> 16) & 0xff;
-	_b[6] = (len >> 8) & 0xff;
-	_b[7] = len & 0xff;
-}
-
-shared_ptr<Node> Host::noteNode(NodeId _id, bi::tcp::endpoint _a, Origin _o, bool _ready, NodeId _oldId)
+void Host::onNodeTableEvent(NodeId _n, NodeTableEventType _e)
 {
-	RecursiveGuard l(x_peers);
-	if (_a.port() < 30300 || _a.port() > 30305)
-		cwarn << "Non-standard port being recorded: " << _a.port();
 
-	if (_a.port() >= /*49152*/32768)
+	if (_e == NodeEntryAdded)
 	{
-		cwarn << "Private port being recorded - setting to 0";
-		_a = bi::tcp::endpoint(_a.address(), 0);
-	}
-
-//	cnote << "Node:" << _id.abridged() << _a << (_ready ? "ready" : "used") << _oldId.abridged() << (m_nodes.count(_id) ? "[have]" : "[NEW]");
-
-	// First check for another node with the same connection credentials, and put it in oldId if found.
-	if (!_oldId)
-		for (pair<h512, shared_ptr<Node>> const& n: m_nodes)
-			if (n.second->address == _a && n.second->id != _id)
+		clog(NetNote) << "p2p.host.nodeTable.events.nodeEntryAdded " << _n;
+		
+		auto n = (*m_nodeTable)[_n];
+		if (n)
+		{
+			RecursiveGuard l(x_sessions);
+			auto p = m_peers[_n];
+			if (!p)
 			{
-				_oldId = n.second->id;
-				break;
+				m_peers[_n] = make_shared<PeerInfo>();
+				p = m_peers[_n];
+				p->id = _n;
 			}
-
-	unsigned i;
-	if (!m_nodes.count(_id))
-	{
-		if (m_nodes.count(_oldId))
-		{
-			i = m_nodes[_oldId]->index;
-			m_nodes.erase(_oldId);
-			m_nodesList[i] = _id;
-		}
-		else
-		{
-			i = m_nodesList.size();
-			m_nodesList.push_back(_id);
+			p->address = n.endpoint.tcp;
+			
+			if (peerCount() < m_idealPeerCount)
+				connect(p);
 		}
-		m_nodes[_id] = make_shared<Node>();
-		m_nodes[_id]->id = _id;
-		m_nodes[_id]->index = i;
-		m_nodes[_id]->idOrigin = _o;
 	}
-	else
+	else if (_e == NodeEntryRemoved)
 	{
-		i = m_nodes[_id]->index;
-		m_nodes[_id]->idOrigin = max(m_nodes[_id]->idOrigin, _o);
+		clog(NetNote) << "p2p.host.nodeTable.events.nodeEntryRemoved " << _n;
+		
+		RecursiveGuard l(x_sessions);
+		m_peers.erase(_n);
 	}
-	m_nodes[_id]->address = _a;
-	m_ready.extendAll(i);
-	m_private.extendAll(i);
-	if (_ready)
-		m_ready += i;
-	else
-		m_ready -= i;
-	if (!_a.port() || (isPrivateAddress(_a.address()) && !m_netPrefs.localNetworking))
-		m_private += i;
-	else
-		m_private -= i;
-
-//	cnote << m_nodes[_id]->index << ":" << m_ready;
-
-	m_hadNewNodes = true;
-
-	return m_nodes[_id];
 }
 
-Nodes Host::potentialPeers(RangeMask<unsigned> const& _known)
+void Host::seal(bytes& _b)
 {
-	RecursiveGuard l(x_peers);
-	Nodes ret;
-
-	auto ns = (m_netPrefs.localNetworking ? _known : (m_private + _known)).inverted();
-	for (auto i: ns)
-		ret.push_back(*m_nodes[m_nodesList[i]]);
-	return ret;
+	_b[0] = 0x22;
```

Comment:
For now this is placeholder for framing; for now, the function is a canonical definition.